tests/ui/resolve/resolve-same-name-struct.rs

```diff
@@ -0,0 +1,33 @@
+//! Test that name resolution works correctly when a struct and its constructor
+//! function have the same name within a nested scope. This checks that the
+//! compiler can distinguish between type names and value names in the same
+//! namespace.
+//!
+//! This is a historical test from early Rust development (circa 2012) when
+//! there was a bug that prevented referring to constructors for structs
+//! nested inside functions from the struct's outer scope.
```

Comment:
```suggestion
```

Same thing; if there is an existing bug then it can be linked, but otherwise it doesn't make anything more clear to say a test didn't pass at some point (which is probably true for the majority of tests when they were added).